### PR TITLE
Place upper limit on tas user validation, remove selected identity notion

### PR DESCRIPTION
**Pull request by cdosborn**
## Description

This changeset places an upper limit on the tas api calls (5sec, in practice the average is about 0.6s), removes all references to selected identities, and makes the user validation and account creation explicit in api/v1/profile (originally the purview of select identity logic)

User validation and account creation happen as a side effect of the serialization of a user's "selected identity" in the api/v1/profile endpoint. Atmosphere used to have the notion of a selected identity for each user. A user's actions would occur against the cloud pointed to by their selected identity. In the frontend, the user would first select a cloud, before they could perform actions against it. All references to a selected identity are vestigial. In jetstream we discovered that changing a user's profile would sometimes take 10-15s, this occurred because the serialization of the selected_identity was checking if the user was valid by querying out to the TAS API. After checking that the user was valid, the service would make sure that the user had identities on all providers where they were permitted. 

Depends on https://github.com/cyverse/clank/pull/277

## Checklist before merging Pull Requests
- [x] Add an entry in the changelog
- [x] Reviewed and approved by at least one other contributor.
- [x] New test(s) included to reproduce the bug/verify the feature
  - There are now tests which ensure create_new_accounts logic is preserved and that the offline validation works (fixed in f839f02)
- [x] New variables supported in Clank

### CODE REVIEW

**Comment by julianpistorius** [comment]
Thanks @cdosborn! I'd like to dig through this a bit more.

File: core/models/user.py [hunk 1]
```diff
@@ -181,7 +181,7 @@ def email_hash(self):
 # Save Hooks Here:
 
 
-def get_or_create_user_profile(sender, instance, created, **kwargs):
```
Comment (julianpistorius) [inline core/models/user.py]: Huh. What's going on here?

File: jetstream/tas_api.py [hunk 2]
```diff
@@ -16,7 +16,8 @@ def tacc_api_post(url, post_data, username=None, password=None):
     # logger.debug("REQ BODY: %s" % post_data)
     resp = requests.post(
         url, post_data,
-        auth=(username, password))
+        auth=(username, password),
+        timeout=5)
```
[the remark this answers is not in the document]
Comment (cdosborn) [reply jetstream/tas_api.py]: Yeah i'll add that

**Comment by cdosborn** [comment]
Just going to push this into the next release

**Comment by julianpistorius** [comment]
> Just going to push this into the next release

We _can_ get it into the  v33 Jetstream release if you want.

**Comment by cdosborn** [comment]
The timeout is the only functional change, since there is a decent amount of refactoring here. I'd prefer for this to ruminate in a release. @calvinmclean's tropo pr should make the experience noticeably better

**Comment by cdosborn** [comment]
@julianpistorius i'm tacking on some tests in a short bit

**Comment by cdosborn** [comment]
@julianpistorius This is being blocked by https://github.com/cyverse/clank/pull/277, once that is :+1: the travis build should greenlight.

**Pull request merged**